### PR TITLE
Add signficiantly more logging to the reporting code

### PR DESCRIPTION
**Pull request by alex**

### CODE REVIEW

File: lib/sources/caseflow/reports.rb [hunk 1]
```diff
@@ -159,8 +159,15 @@ def main(argv)
 
   report_cases = Caseflow::Reports::ConcurrentArray.new
   Parallel.each(vacols_cases, in_threads: 4, progress: "Checking VBMS") do |vacols_case|
-    if report.should_include(vacols_case)
-      report_cases << vacols_case
+    begin
+      if report.should_include(vacols_case)
+        Rails.logger.info "event=report.case.found bfkey=#{vacols_case.bfkey}"
```
Comment (paultag) [inline lib/sources/caseflow/reports.rb]: is this guy threadsafe? I wonder if we'll start mashing up the output fd (I mean, it's rails so I'm assuming yes, but just checking)

Comment (paultag) [inline lib/sources/caseflow/reports.rb]: Oh jeez, http://log.kares.org/2011/04/railslogger-is-not-threadsafe.html

>  At least not completely, one might experience strange logging issues especially in JRuby threadsafe! environments: logs show up fine in the beginning but after a while it seems as if logging stopped working and only errors show up.

Comment (alex) [inline lib/sources/caseflow/reports.rb]: While obnoxious, it looks like this is only true with the `silence` method that we're not using?

Comment (paultag) [inline lib/sources/caseflow/reports.rb]: Looking into it more, silence isn't being used, should be fine.

Comment (paultag) [inline lib/sources/caseflow/reports.rb]: Yeah, just finished groking and came to the same conclusion myself

**Comment by paultag** [comment]
LGTM

**Pull request merged**